### DISCUSSION TITLE
Renew auth

### DISCUSSION THREAD

**Discussion by mars3142** [GodotNuts/GodotFirebase#175] [Q&A] [answered]
The google login is valid for around 1h. Do I need to reauthenticate the user by myself or will this be done automatically, after the JWT is invalid? 

And does this plugin persists the last JWT or do I need to implement a relogin after a new app start?

**Comment by fenix-hub** [comment, marked as answer]
Hi @mars3142 
Currently the JWT obtained with `get_google_oauth()` is only used to exchange it with a Google Id Token. That's the one that is being refreshed automatically by our plugin and the one used to make requests.
So the first JWT you copy from Google OAuth2 page is not stored either refreshed, since it is used just as a one-time token.
If you want to use it again (eventually because you log out of the current user, or just closed the app), you will need to store it somewhere. If it expires it is not a big issue, since if you are already authenticated it won't be necessary anyway, and if you need a new one there is no automatic procedure currently, but you always have to call the `get_google_oauth()` function and copy-paste the JWT.
If you think that storing it would be useful for futher operations, please let me know and I can work on it.
I hope I answered correctly to your questions :)
